pkg/operator/controller/controller_service_monitor.go

```diff
@@ -54,7 +64,11 @@ func desiredServiceMonitor(ci *operatorv1.IngressController, svc *corev1.Service
 						"openshift-ingress",
 					},
 				},
-				"selector": map[string]interface{}{},
+				"selector": map[string]interface{}{
+					"matchLabels": map[string]interface{}{
+						manifests.OwningIngressControllerLabel: ci.Name,
+					},
+				},
 				"endpoints": []map[string]interface{}{
```

Comment:
You need this change or else `serviceMonitorChanged` may panic on old resources and return false positives on updated resources:  https://github.com/openshift/cluster-ingress-operator/commit/4c56ec1fcfc538eb1801c914d807eec4d5e33775#diff-b3b3514d6621705093f820e6f9feead0R70-R80

Reply:
Fixed, thanks